text/0040-pagination-convention.md

```diff
@@ -0,0 +1,118 @@
+- Title: Pagination Convention
+- Start Date: 2021-05-10
+- Specification PR: [#40](https://github.com/meilisearch/specifications/pull/40)
+- MeiliSearch Tracking-Issues:
+
+# Pagination Convention
+
+## 1. Feature Description and Interaction
+
+### I. Summary
+
+Pagination allows the content to be viewed in the form of a sequence in order to facilitate the browsing of its content by not overloading the number of elements to be viewed for the user. Pagination also has the advantage of easily providing the total number of items that remain to be traversed by not overloading the network with all the content that can be loaded.
+
+### II. Motivation
+
+This specification will be used as a convention to implement simple and standard paginations to be set up on API routes such as the list of updates for example. Since the pagination of search results is intrinsically linked to technical implementations of the search engine, this specific case is detailed in its own specification.
+
+### III. Additional Materials
+N/A
+
+### IV. Explanation
+
+##### Requirements
+
+- ✅ Each paginable route must return the total number of items it can render to the user at the end and the current number of items returned in the response. E.g. `total` and `size`. In the case of multiples pages, all other pages except the last one will have a `size = limit`.
```

Comment:
I'm not sure if you are suggesting this becomes a default behavior and not an activable option?

Reply:
Can you please tell me which part makes you think of an option that can be activated on demand? Having `total` and `size` can be the default behavior on standard paginated resources in the response body. `size` represents the number of items in the search result for the current page and `total` represents the total number of items to be segmented on the whole pages. `offset` and `limit` parameters are used to move between pages.